prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,202 @@
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for deleting a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to patch.
+        body: A dictionary representation of a Kubernetes V1Job
+            specification.
+        namespace: The Kubernetes namespace to patch this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Raises:
+        ValueError: if `job_name` is `None`
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    return api_client.patch_namespaced_job(
+        name=job_name, namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+def read_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for reading a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to read.
+        namespace: The Kubernetes namespace to read this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Raises:
+        ValueError: if `job_name` is `None`.
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    return api_client.read_namespaced_job(
+        name=job_name, namespace=namespace, **kube_kwargs
+    )
+
+
+@task
+def replace_namespaced_job(
+    body: dict,
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for replacing a namespaced Kubernetes job.
+
+    Args:
+        body: A dictionary representation of a Kubernetes V1Job
+            specification
+        job_name: The name of a job to replace.
+        namespace: The Kubernetes namespace to replace this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    return api_client.replace_namespaced_job(
```

Comment:
I guess Kubernetes API should be something like Docker API so I think we may still want the function async and wrap this call with run_sync_in_worker_thread like https://github.com/PrefectHQ/prefect-docker/blob/main/prefect_docker/containers.py#L237